### PR TITLE
fix(pg): prevent unhandled promise rejection

### PR DESCRIPTION
**Pull request by glasseyes42**
Fixes: elastic/apm-agent-nodejs#1829

### Checklist

- [x] Implement code
- [x] Add tests
- [x] Commit message follows [commit guidelines](https://github.com/elastic/apm-agent-nodejs/blob/master/CONTRIBUTING.md#commit-message-guidelines)

### CODE REVIEW

**Comment by astorm** [comment]
@glasseyes42 Our CI seems unhappy about ... something.  I'm going to follow up with our tools team to figure out what's going on.  While I do that -- could you confirm you've signed https://www.elastic.co/contributor-agreement?

**Comment by astorm** [comment]
jenkins run the tests please

**Comment by astorm** [comment]
jenkins run the tests

**Comment by glasseyes42** [comment]
@astorm yes, can confirm I've signed the agreement. I believe its one of the checks that has passed already 👍

**Comment by astorm** [comment]
jenkins run the tests

File: lib/instrumentation/modules/pg.js [hunk 1]
```diff
@@ -103,7 +103,15 @@ function patchClient (Client, klass, agent, enabled) {
...
+            query = query
+              .then(function (result) {
+                end()
+                return result
+              })
+              .catch(function (err) {
+                end()
+                throw err
+              })
           } else {
```
Comment (astorm) [inline lib/instrumentation/modules/pg.js:115]: Prior to this PR this branch of the conditional would add our "end the span" callback to the end of the promise chain to ensure it was called.  I see that this PR expands on that to make sure any unhandled promise rejections are caught (avoid a fatal error) and then the error is re-thrown to ensure legitimate errors are now swallowed up.

**Review by astorm — APPROVED** [review]

**Pull request merged**